flasgger/utils.py

```diff
@@ -258,19 +258,29 @@ def wrapper(*args, **kwargs):
     return decorator
 
 
-def __replace_ref(schema, relative_path):
+def __replace_ref(schema, relative_path, swag):
     """ TODO: add dev docs
 
     :param schema:
     :param relative_path:
+    :param swag:
     :return:
     """
     absolute_path = os.path.dirname(sys.argv[0])
     new_value = {}
     for key, value in schema.items():
         if isinstance(value, dict):
-            new_value[key] = __replace_ref(value, relative_path)
+            new_value[key] = __replace_ref(value, relative_path, swag)
         elif key == '$ref':
+            # see:
+            # https://swagger.io/docs/specification/describing-request-body/
+            if len(value) > 2 and value.startswith('#/'):  # $ref is local
```

Comment:
why don't you consider get the components from different files as well?

Reply:
I fixed an issue I had ;-)
Have you got an example with multiple files that doesn't work ? Because, from what I read, I think that accessing files is done elsewhere.